junit5-jupiter-starter-bazel/BUILD

```diff
@@ -0,0 +1,19 @@
+load("//:junit5.bzl", "java_junit5_test")
+
+java_library(
+    name = "calculator",
+    srcs = [
+        "src/main/java/com/example/project/Calculator.java",

```

Comment:
@marcphilipp For future reference, I understand that _recursive_ globs (those containing `**`) are discouraged in Bazel for performance and maintainability reasons. Non-recursive globs are fine. (See [this entry](https://docs.bazel.build/versions/master/skylark/build-style.html#globs) in the BUILD file style guide for more info if you're interested.)

Reply:
Interesting! Is there a reasoning how a `BUILD` file per package is more maintainable? I find this recommendation rather strange for a regular Java project. Why should a certain set of classes be compiled differently than those in another package? 🤔

Reply:
> Interesting! Is there a reasoning how a BUILD file per package is more maintainable? I find this recommendation rather strange for a regular Java project. Why should a certain set of classes be compiled differently than those in another package? :thinking:

Now, I don't know _much_ about Bazel yet, but I believe that the reason it's more maintainable and that they have this strange recommendation is two-fold:

1. I don't know for certain, so this is just an educated guess of mine, but I believe that in very large code bases like Google's, it's important that engineers can tell at a glance (or reasonably quickly) what a given BUILD file builds. If they have to traverse to an inner directory which may be _many_ levels down just to figure out what gets built by a `java_library` call (which they'd have to do if the call uses a recursive glob like `src/main/java/**.java`), then it could lead to engineers wasting time and effort.

    Likewise, I believe that Bazel itself has to spend more time just to discover all classes pointed to by the recursive glob, a task which I believe is difficult for Bazel to cache, meaning that each time a Bazel build is run, it would have to find the files pointed to by the recursive glob all over again.

2. By my understanding, it's not a matter of having certain classes be compiled differently, but rather it's a matter of having smaller, independent _units_ of classes that can be compiled in parallel.

    So you may or may not know that Bazel is designed to build stuff very quickly, and one way it does so is to build things in parallel. But to do so, it needs some help from the user.

    If the user provides just one BUILD file that says "please compile all .java files underneath `src/main/java` for me", then that's fine, Bazel can do it's job and call `javac` (or whatever equivalent it uses). However, it's forced to compile the files _sequentially_, because it doesn't know how to split up the job of compiling the .java files into smaller units that can be sent to multiple `javac`s and compiled in parallel.

    Therefore, the Bazel maintainers [recommend](https://docs.bazel.build/versions/master/bazel-and-java.html#build-files) that multiple BUILD files be provided: one per directory that directly holds one or more .java files. This way, by my limited understanding, Bazel can read each BUILD file in parallel, and therefore it can compile each directory of .java files in parallel too, meaning (on a multi-core machine) that the _entire_ set of .java files is compiled a lot more quickly.

    A side-effect of this recommendation is that users can compile just _certain_ directories if they so wish, as long as those directories have BUILD files. (This is important in places like Google where everything is in a humongous monorepo and building everything at once is impractical.)

I hope this sheds some light for you, @marcphilipp. :)